### ISSUE TITLE
bug: `diff` children, the vnodes can includes `null` node

### ISSUE THREAD

**Issue by cbbfcd** [aidenybai/million#209]
**Describe the bug**
<img width="1630" alt="image" src="https://user-images.githubusercontent.com/20987775/175772829-f2ef3a55-b4f1-42b5-bbde-ddea59e0b75e.png">

<img width="1151" alt="image" src="https://user-images.githubusercontent.com/20987775/175772865-f579db61-ea03-44d1-b184-d571ebf819c7.png">

so, the element can be removed, it's not work as excepted.

if we filter the `vnodes` in my demo, it work perfect.
<img width="843" alt="image" src="https://user-images.githubusercontent.com/20987775/175772980-95bb35a8-fcb1-46b2-9f13-b95aa9e13fac.png">

but, this causes the order of vnodes to be out of order, i try to find a nother way to resolve it. what's your opinion?
**To Reproduce**

**Expected behavior**

**Screenshots**

**Device (please complete the following information):**

**Additional context**

**Comment by aidenybai** [comment]
I think in this case Million defers the responsibility of children normalization to the `h` function.

This way, computation to remove nully values in VNode does not occur during the runtime, but during compile time

**Comment by aidenybai** [comment]
Does #210 fix for Wouter?

**Comment by cbbfcd** [comment]
if the runtime does not handle null children-vnode, there will be problems like this.

if the `vnode children` is :
```js
const vnodeChildren = [
{ tag: 'a', ...anyOthers },
{ tag: 'div', ...anyOthers },
null,
{ tag: 'will-be-deleted', ...anyOthers }
]
```

when run into the `diff` function:
```js
for (let i = vnodeChildren - 1;  i >= 0 ; --i ) {
  // what's the problem.
  effects = diff(el.childNodes.item(i), newVnodeChildren[i], oldVNodeChildren(i));
}
```

`el.childNodes` will filter the `null` child, so `el.childNodes.item(3)` is `empty` and `el.childNodes.item(2)` matched newVnode is `null`, so the third element will be remove.

**Comment by cbbfcd** [comment]
> Does #210 fix for Wouter?

I found this problem through the `wouter` demo you provided. When the route address is `users/1`, after clicking the button twice, the rendered `Route` component will disappear.

i debug for this problem,

it is found that the root of the problem is that vnode.children contains null, and el.childNodes does not contain null, so there may be problems when executing diff, and `remove` is executed on the wrong object

**Comment by aidenybai** [comment]
I see, I'll make some changes and merge your PR. Thanks for the explanation

**Comment by aidenybai** [comment]
I'll be adding some more fixed to this PR to fix #203 -- expect a merge by the end of this week!
